Close all file descriptors except for the standard ones not to hang
dpkg-reconfigure

Index: dictd/dictd.c
===================================================================
--- dictd.orig/dictd.c	2009-03-08 23:29:36.000000000 +0100
+++ dictd/dictd.c	2009-03-08 23:30:43.000000000 +0100
@@ -1485,6 +1485,10 @@
       { 0,                  0, 0, 0  }
    };
 
+   /* close all file descriptors except for the standard ones */
+   for (i=getdtablesize()-1; i > 2; --i)
+ 	close(i);
+
    set_umask ();
    init (argv[0]);
 
